cmake/HiOpCheckGitSubmodules.cmake

```diff
@@ -1,30 +1,43 @@
 find_package(Git QUIET)
-if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
-  # Update submodules as needed
-  message(STATUS "Submodule update")
-  execute_process(
-    COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    RESULT_VARIABLE GIT_SUBMOD_RESULT
-  )
-  if(NOT GIT_SUBMOD_RESULT EQUAL "0")
-    message(
-      FATAL_ERROR
-        "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules manually"
+if(GIT_FOUND)
+  if(EXISTS "${PROJECT_SOURCE_DIR}/.git")
+    # Update submodules as needed
+    message(STATUS "Submodule update")
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      RESULT_VARIABLE GIT_SUBMOD_RESULT
     )
-  endif()
-endif()
+    if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+      message(
+        FATAL_ERROR
+        "'git submodule update --init --recursive' failed with ${GIT_SUBMOD_RESULT}. "
+        "Please checkout submodules manually with the folowing command and run cmake again:\n"
+        "\t$ git submodule update --init --recursive\n"
+        )
```

Comment:
Is the "fatal error" the best way to go about this? Perhaps just disabling Eigen would suffice. One can still build other components of HiOp. My understanding is that fatal error would prevent building HiOp altogether.

Reply:
> Is the "fatal error" the best way to go about this? Perhaps just disabling Eigen would suffice. One can still build other components of HiOp. My understanding is that fatal error would prevent building HiOp altogether.

we can go with a warning, but that is really an error on the user ends because `.git` exists but `git submodule` fails